frontend/src/pages/modelCatalog/ModelCatalogCoreLoader.tsx

```diff
@@ -1,26 +1,51 @@
 import * as React from 'react';
+import { Outlet } from 'react-router';
 import { conditionalArea, SupportedArea } from '~/concepts/areas';
 import ApplicationsPage from '~/pages/ApplicationsPage';
 import TitleWithIcon from '~/concepts/design/TitleWithIcon';
 import { ProjectObjectType, typedEmptyImage } from '~/concepts/design/utils';
+import {
+  ModelCatalogContext,
+  ModelCatalogContextProvider,
+} from '~/concepts/modelCatalog/context/ModelCatalogContext';
 import EmptyModelCatalogState from './EmptyModelCatalogState';
 
+type ApplicationPageProps = React.ComponentProps<typeof ApplicationsPage>;
+
+type ApplicationPageRenderState = Pick<
+  ApplicationPageProps,
+  'emptyStatePage' | 'empty' | 'headerContent'
+>;
 const ModelCatalogCoreLoader: React.FC = conditionalArea(
   SupportedArea.MODEL_CATALOG,
   true,
 )(() => {
-  const renderStateProps = {
-    empty: true,
-    emptyStatePage: (
-      <EmptyModelCatalogState
-        testid="empty-model-catalog-state"
-        title="Request access to model catalog"
-        description="To request access to model catalog, contact your administrator."
-        headerIcon={() => <img src={typedEmptyImage(ProjectObjectType.registeredModels)} alt="" />}
-      />
-    ),
-    headerContent: null,
-  };
+  const { modelCatalogSources } = React.useContext(ModelCatalogContext);
+  let renderStateProps: ApplicationPageRenderState & { children?: React.ReactNode };
+  if (modelCatalogSources.length === 0) {
+    renderStateProps = {
+      empty: true,
+      emptyStatePage: (
+        <>
+          <EmptyModelCatalogState
+            testid="empty-model-catalog-state"
+            title="Request access to model catalog"
+            description="To request access to model catalog, contact your administrator."
+            headerIcon={() => (
+              <img src={typedEmptyImage(ProjectObjectType.registeredModels)} alt="" />
+            )}
+          />
+        </>
+      ),
+      headerContent: null,
+    };
+  } else {
+    return (
+      <ModelCatalogContextProvider>
+        <Outlet />
+      </ModelCatalogContextProvider>
+    );
+  }
```

Comment:
We can simplify this a little - this pattern came from other CoreLoaders where we had more complicated logic and multiple possible `renderStateProps` values. Here though there's no reason not to just `return <ApplicationsPage ...` right in this `if (modelCatalogSources.length === 0) {` condition. and then you can remove the `else` and just return the provider/outlet after the if block.